views/svg/ResponsiveSvg.svelte

```diff
@@ -4,8 +4,22 @@
   export let aspectRatio;
 </script>
 
-<div
-  class="q-choropleth-svg-container"
-  style="padding-top: {round(100 / aspectRatio)}%">
+<div class="svg-container" style="padding-top: {round(100 / aspectRatio)}%">
   <slot />
 </div>
+
+<style>
+  .svg-container {
+    position: relative;
+    margin-top: 8px;
+    margin-bottom: 8px;
+  }
+
+  .svg-container :global(svg) {
+    display: block;
+    position: absolute;
+    top: 0;
+    width: 100%;
+    height: 100%;
+  }
+</style>
```

Comment:
I'm a little confused here. Could you explain, why there're two separate blocks with different styles for the same class `.svg-container`? And isn't the `:global(svg)` affecting all the svg on the site? Maybe all of this is fine and needed, but it helps my understanding.

Reply:
Yes sure. The first block appies to all elements with class svg-container within this svelte component. The second block applies to all svg tags with parent with class svg-container within this svelte component. :global is required to be used because this component uses </slot>. The svg is not defined within the component, therefor svelte would automatically remove the css styles if we would write it like this: .svg-container svg. The only way to apply styles to the svg is by using :global(). This forces svelte to keep the svg styles.

See also documentation about it on [svelte.dev ](https://svelte.dev/docs)and [this stackoverflow answer](https://stackoverflow.com/questions/46086483/how-to-apply-styles-to-slot-element-in-svelte):

<img width="1075" alt="Screenshot 2021-05-21 at 15 05 35" src="https://user-images.githubusercontent.com/1810384/119142236-882c1700-ba46-11eb-908e-518d7569aa61.png">

The styles are not applied globally to every svg element on the page. Only to all svg tags with parent with class svg-container within this component.

Reply:
Ah, I see. I wasn't aware of that and I also forgot about the `<slot />`. Very cool stuff. Thanks for clearing it up for me!

Reply:
If you have the svelte for VS Code extension installed the editor will let you know if you have unused CSS rules. It can be quite useful. Let me know if you see other things, which are not clear. 

<img width="605" alt="Screenshot 2021-05-21 at 15 36 06" src="https://user-images.githubusercontent.com/1810384/119145906-659bfd00-ba4a-11eb-8e6c-be8bba43a9df.png">